dht/token.hh

```diff
@@ -23,7 +23,10 @@
 
 #include "bytes.hh"
 #include "utils/managed_bytes.hh"
+#include <seastar/net/byteorder.hh>
```

Comment:
I don't think this is needed in this patch.

Reply:
Without it the compilation fails with:
```
In file included from dht/token.cc:27:
./dht/token.hh:125:12: error: ‘data_type’ does not name a type
dht/token.cc:184:1: error: no declaration matches ‘data_type dht::token::get_token_validator()’
dht/token.cc:184:1: note: no functions named ‘data_type dht::token::get_token_validator()’In file included from dht/token.cc:27:
./dht/token.hh:42:7: note: ‘class dht::token’ defined here
```

Reply:
Removed.